### PR TITLE
chore: execute "mark as read" mutation only when notification is unread

### PR DESCRIPTION
**Pull request by dimatretyak**
This PR resolves [] 

### PR Checklist

- [ ] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [x] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/Scenes/Activity/ActivityItem.tests.tsx [hunk 1]
```diff
@@ -153,6 +149,22 @@ describe("ActivityItem", () => {
...
+      Notification: () => notification,
+    })
+    await flushPromiseQueue()
+
+    fireEvent.press(getByText("Notification Title"))
+
+    await flushPromiseQueue()
+
+    const recentOperations = mockEnvironment.mock.getAllOperations()
+    expect(recentOperations).toHaveLength(0)
```
Comment (dimatretyak) [inline src/app/Scenes/Activity/ActivityItem.tests.tsx:165]: @gkartalis do you think it's a good idea to check if any mutations are performed in this way?

Comment (dimatretyak) [reply src/app/Scenes/Activity/ActivityItem.tests.tsx:165]: or is it better to do such a check?

```javascript
expect(() => mockEnvironment.mock.getMostRecentOperation()).toThrow(
  "There are no pending operations in the list"
)
```

Comment (gkartalis) [reply src/app/Scenes/Activity/ActivityItem.tests.tsx:165]: I believe that this one is better ☝️

**Review by gkartalis — APPROVED** [review]
also great followup, 🤯 I totally missed that

File: src/app/Scenes/Activity/ActivityItem.tests.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/app/Scenes/Activity/ActivityItem.tests.tsx:165]: add changes

Comment (gkartalis) [reply src/app/Scenes/Activity/ActivityItem.tests.tsx]: not sure if it is like this or :
```
expect(mockEnvironment.mock.getMostRecentOperation()).toThrowError( "There are no pending operations in the list" )
```

Comment (dimatretyak) [reply src/app/Scenes/Activity/ActivityItem.tests.tsx]: in this case we got the following error
image

**Pull request merged**